tools/REALTEK_RTL8195AM.py

```diff
@@ -27,108 +27,98 @@
+                match = re.match(r'^' + section + \
+                        r'\s+0x0{,8}(?P<addr>[0-9A-Fa-f]{8})\s+.*$', line)
+            elif toolchain in ["ARM_STD", "ARM", "ARM_MICRO"]:
+                # Memory Map of the image
+                #   Load Region LR_DRAM (Base: 0x30000000, Size: 0x00006a74, Max: 0x00200000, ABSOLUTE)
+                #     Execution Region IMAGE2_TABLE (Base: 0x30000000, Size: 0x00000018, Max: 0xffffffff, ABSOLUTE, FIXED)
+                #     Base Addr    Size         Type   Attr      Idx    E Section Name        Object
+                #     0x30000000   0x00000004   Data   RO         5257    .image2.ram.data    rtl8195a_init.o
                 match = re.match(r'^.*Region\s+' + section + \
                         r'\s+\(Base: 0x(?P<addr>[0-9A-Fa-f]{8}),\s+Size:.*\)$', line)
-                if match:
-                    ret = match.group("addr")
-                    break
-    elif toolchain == "IAR":
-        #   Section                 Kind        Address     Size  Object
-        #   -------                 ----        -------     ----  ------
-        # "A3":                                           0x8470
-        #   IMAGE2                           0x10006000   0x5d18  <Block>
-        #     .ram_image2.text               0x10006000   0x5bbc  <Block>
-        #       .rodata             const    0x10006000     0x14  retarget.o [17]
-        with open(mapfile, 'r') as infile:
-            # Search area to parse
-            for line in infile:
-                if line.startswith('  Section  '):
-                    break
-            # Search section(block)
-            for line in infile:
+            elif toolchain == "IAR":
+                #   Section                 Kind        Address     Size  Object
+                #   -------                 ----        -------     ----  ------
+                # "A3":                                           0x8470
+                #   IMAGE2                           0x10006000   0x5d18  <Block>
+                #     .ram_image2.text               0x10006000   0x5bbc  <Block>
+                #       .rodata             const    0x10006000     0x14  retarget.o [17]
                 match = re.match(r'^\s+' + section + \
-                        r'\s+0x(?P<addr>[0-9A-Fa-f]{8})\s+.*<Block>$', line)
-                if match:
-                    ret = match.group("addr")
-                    break
-    else:
-        print "[ERROR] unsupported toolchain " + toolchain
-        sys.exit(-1)
-
-    if not ret:
+                        r'\s+0x(?P<addr>[0-9A-Fa-f]{8})\s+0x(?P<size>[0-9A-Fa-f]+)\s+.*<Block>$', line)
+            if match:
+                ret1 = match.group("addr")
+                try:
+                    ret2 = match.group("size")
+                except IndexError:
+                    pass
+                break
+
+    if not ret1:
         print "[ERROR] cannot find the address of section " + section
         return 0
```

Comment:
This return value is not of the same type as the dict returned below. You might want to make this `{}` instead of `0`, so that the types match.